### PR TITLE
[NFC][AArch64][SVE] Rename variables in partial reduction lowering functions

### PR DESCRIPTION
**Pull request by JamesChesterman**

### CODE REVIEW

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 1]
```diff
@@ -21739,73 +21739,71 @@ SDValue tryLowerPartialReductionToDot(SDNode *N,
...
 
-  auto ExtA = MulOp->getOperand(0);
-  auto ExtB = MulOp->getOperand(1);
+  auto ExtMulOp1 = B->getOperand(0);
+  auto ExtMulOp2 = B->getOperand(1);
 
-  if (!ISD::isExtOpcode(ExtA->getOpcode()) ||
-      !ISD::isExtOpcode(ExtB->getOpcode()))
+  if (!ISD::isExtOpcode(ExtMulOp1->getOpcode()) ||
+      !ISD::isExtOpcode(ExtMulOp2->getOpcode()))
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21751]: It probably helps to combine some of the conditions together, e.g.:
```suggestion
  SDValue Op2 = N->getOperand(2);
  if (Op2->getOpcode() != ISD::MUL || 
      !ISD::isExtOpcode(Op2->getOperand(0)->getOpcode()) ||
      !ISD::isExtOpcode(Op2->getOperand(1)->getOpcode()))
    return SDValue();
```

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 2]
```diff
@@ -21739,73 +21739,71 @@ SDValue tryLowerPartialReductionToDot(SDNode *N,
...
-  bool AIsSigned = ExtA->getOpcode() == ISD::SIGN_EXTEND;
-  bool BIsSigned = ExtB->getOpcode() == ISD::SIGN_EXTEND;
+  bool MulOp1IsSigned = ExtMulOp1->getOpcode() == ISD::SIGN_EXTEND;
+  bool MulOp2IsSigned = ExtMulOp2->getOpcode() == ISD::SIGN_EXTEND;
 
-  auto A = ExtA->getOperand(0);
-  auto B = ExtB->getOperand(0);
-  if (A.getValueType() != B.getValueType())
+  auto MulOp1 = ExtMulOp1->getOperand(0);
+  auto MulOp2 = ExtMulOp2->getOperand(0);
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21757]: Now that we know this is a recognised 'dot' operation we can define the operands and sub-operands with their appropriate names:
```
  SDValue Acc = N->getOperand(1);
  SDValue Mul = N->getOperand(2);
  SDValue ExtMulOpLHS = Mul->getOperand(0);
  SDValue ExtMulOpRHS = Mul->getOperand(1);
```

I realise this is defining N->getOperand(2) as both `Mul` and `Op2`, but in your later patch `Mul` will have to be generated (by multiplying `Op2` with `splat(1)`).

nit: For a binary operation like 'mul', it's common to use `LHS` and `RHS` for the operands.
nit: Only use `auto` when it's clear from the context what the type is. In this case, `SDValue Acc` is more appropriate than `auto Acc`.

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 3]
```diff
@@ -21739,73 +21739,71 @@ SDValue tryLowerPartialReductionToDot(SDNode *N,
...
+  if (MulOp1.getValueType() != MulOp2.getValueType())
     return SDValue();
 
-  EVT ReducedType = N->getValueType(0);
-  EVT MulSrcType = A.getValueType();
+  EVT AVT = N->getValueType(0);
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21761]: Isn't this the result type of the partial reduction itself? If so, I would just name this `ReducedVT`.

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 4]
```diff
@@ -21822,32 +21820,32 @@ SDValue tryLowerPartialReductionToWideAdd(SDNode *N,
...
+  auto A = N->getOperand(1);
+  auto ExtB = N->getOperand(2);
 
-  EVT AccVT = Acc.getValueType();
-  EVT AccElemVT = AccVT.getVectorElementType();
+  EVT AVT = A.getValueType();
+  EVT AElemVT = AVT.getVectorElementType();
 
-  if (ExtInput.getValueType().getVectorElementType() != AccElemVT)
+  if (ExtB.getValueType().getVectorElementType() != AElemVT)
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21829]:
```
  if (!ISD::isExtOpcode(N->getOperand(2).getOpcode())
    return SDValue();
  SDValue Acc = N->getOperand(1);
  SDValue Ext = N->getOperand(2);
  EVT AccVT = Acc.getValueType();
  EVT ExtVT = Ext.getValueType();
  if (!ExtVT.getVectorElementType() != AccVT.getVectorElementType())
    return SDValue();
```

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp [hunk 5]
```diff
@@ -21822,32 +21820,32 @@ SDValue tryLowerPartialReductionToWideAdd(SDNode *N,
...
-  unsigned ExtInputOpcode = ExtInput->getOpcode();
-  if (!ISD::isExtOpcode(ExtInputOpcode))
+  unsigned ExtBOpcode = ExtB->getOpcode();
+  if (!ISD::isExtOpcode(ExtBOpcode))
     return SDValue();
 
-  auto Input = ExtInput->getOperand(0);
-  EVT InputVT = Input.getValueType();
+  auto B = ExtB->getOperand(0);
+  EVT BVT = B.getValueType();
```
Comment (sdesmalen-arm) [inline llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21837]:
```suggestion
  SDValue ExtOp = Ext->getOperand(0);
  EVT ExtOpVT = ExtOp.getValueType();
```

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 1 above)
Comment (JamesChesterman) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp]: Done

**Review by sdesmalen-arm — APPROVED** [review]

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 2 above)
Comment (JamesChesterman) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp]: Done, also replaced `MulOp1` and `MulOp2` with `MulOpLHS` and `MulOpRHS`. Also removed the `auto` keyword where it is more appropriate to just put the type.

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 3 above)
Comment (JamesChesterman) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21761]: Done

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 5 above)
Comment (JamesChesterman) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp]: Done

File: llvm/lib/Target/AArch64/AArch64ISelLowering.cpp (hunk 4 above)
Comment (JamesChesterman) [reply llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:21829]: Done

**Pull request merged**